app/controllers/hosts_controller.rb

```diff
@@ -737,6 +738,16 @@ def find_resource
     @host
   end
 
+  def multiple_with_filter?
+    params.key?(:search)
+  end
+
+  def multiple_filter
+    if multiple_with_filter?
+      params[:search].blank? ? _("Empty") : params[:search]
+    end
+  end
```

Comment:
`host_helper` is too large, hence I've extract this logic into `hosts_and_hostgroup_helper`